src/analysis/convex/topology.lean

```diff
@@ -138,6 +139,38 @@ lemma convex.add_smul_mem_interior {s : set E} (hs : convex s)
   x + t • y ∈ interior s :=
 by { convert hs.add_smul_sub_mem_interior hx hy ht, abel }
 
+open affine_map
+
+/-- If we dilate a convex set about a point in its interior by a scale `t > 1`, the interior of
+the result contains the original set. -/
+lemma convex.subset_interior_image_homothety_of_one_lt
```

Comment:
Can you generalize that to `balanced ℝ s`? I think you can split it into `convex.balanced` and `balanced.subset_interior_image_homothety_of_one_lt`.

Reply:
This is a good point and I agree the generalisation is worth having.

However, I don't see any API connecting `convex` and `balanced` (or the "star shaped" affine version I guess) so I think I'd rather just note this generalisation as a TODO in the doc string so that I don't have to build all of this just for this one result.

Of course if anyone has strong feelings, I'll happily carry out the generalisation.

Reply:
I'll add a comment to the doc string but leave this conversation unresolved until I hear more.

Reply:
That's fine too! Do you not think however that your proof translates more or less to a proof of `balanced.subset_interior_image_homothety_of_one_lt` already? If so, you migh as well prove it, even though that duplicates some code.

Reply:
The proof translates directly. I'm not worried about that. It's the missing API that concerns me.

Reply:
Nevermind, `convex s` doesn't imply `balanced s` (you also need `0 ∈ A` and `∀ x ∈ A, -x ∈ A`), so your proof is more general than what I suggested. Still I think `balanced.subset_interior_image_homothety_of_one_lt` is worth adding in this PR.